work-item-comments.go

```diff
@@ -68,22 +68,26 @@ func (c *WorkItemCommentsController) Create(ctx *app.CreateWorkItemCommentsConte
 
 // List runs the list action.
 func (c *WorkItemCommentsController) List(ctx *app.ListWorkItemCommentsContext) error {
+	offset, limit := computePagingLimts(ctx.PageOffset, ctx.PageLimit)
 	return application.Transactional(c.db, func(appl application.Application) error {
 		_, err := appl.WorkItems().Load(ctx, ctx.ID)
 		if err != nil {
-			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrUnauthorized(err.Error()))
-			return ctx.NotFound(jerrors)
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
 		}
 
-		res := &app.CommentArray{}
+		res := &app.CommentList{}
 		res.Data = []*app.Comment{}
 
-		comments, err := appl.Comments().List(ctx, ctx.ID)
+		comments, tc, err := appl.Comments().List(ctx, ctx.ID, &offset, &limit)
+		count := int(tc)
 		if err != nil {
-			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrUnauthorized(err.Error()))
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInternal(err.Error()))
```

Comment:
You can replace these two lines with `return jsonapi.JSONErrorResponse(ctx, err)`